src/System.Runtime.Extensions/src/System/IO/StringWriter.cs

```diff
@@ -178,8 +183,13 @@ public override void WriteLine(ReadOnlySpan<char> buffer)
             WriteLine();
         }
 
-        #region Task based Async APIs
+        public override void WriteLine(StringBuilder value)
```

Comment:
All of these overloads should have a check like the one here:
https://github.com/dotnet/corefx/pull/30667/files#diff-062fa053ba9ce2bdd08a5926cfd4fccaR169

The problem is that someone may have already derived from StringWriter and overridden the core methods to do something different.  If we then don't delegate to those from this new overload, we could start doing something wrong when the new method is used on their derived type.  Since we don't have an efficient way to check for that specific condition, we instead just check whether this is a concrete StringWriter or something derived from it.

Reply:
clear!

Reply:
done